### PR TITLE
Allow querying Bluetooth service status on iOS (Apple)

### PR DESCRIPTION
**Pull request by JeroenWeener**
This PR updates the Apple implementation, fixing an issue where the Bluetooth service status was never resolved properly. To this end it
- Makes `checkServiceStatus` async
- Keeps track of `PermissionStrategy` in `PermissionManager` (hold the reference to guarantee that the callback fires)

Additionally, it updates the displayed permission in the example app, as these were out of sync with the newest permissions.

This PR is part of feature #773.

## Pre-launch Checklist

- [x] I made sure the project builds.
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is does not need version changes.
- [x] I updated `CHANGELOG.md` to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I rebased onto `main`.
- [ ] I added new tests to check the change I am making, or this PR does not need tests.
- [x] I made sure all existing and new tests are passing.
- [x] I ran `dart format .` and committed any changes.
- [x] I ran `flutter analyze` and fixed any errors.

[Contributor Guide]: https://github.com/Baseflow/flutter-permission-handler/blob/master/CONTRIBUTING.md
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

File: permission_handler_apple/ios/Classes/strategies/BluetoothPermissionStrategy.m [hunk 1]
```diff
@@ -23,23 +24,28 @@ - (void)initManagerIfNeeded {
...
     [self initManagerIfNeeded];
+    
+    _serviceStatusHandler = completionHandler;
+    _requestedPermission = permission;
+}
+
+- (void)handleCheckServiceStatusCallback {
     if (@available(iOS 10, *)) {
-        return [_centralManager state] == CBManagerStatePoweredOn ? ServiceStatusEnabled : ServiceStatusDisabled;
+        ServiceStatus serviceStatus = [_centralManager state] == CBManagerStatePoweredOn ? ServiceStatusEnabled : ServiceStatusDisabled;
```
Comment (mvanbeusekom) [inline permission_handler_apple/ios/Classes/strategies/BluetoothPermissionStrategy.m:43]: I am wondering if this really covers all situations. If the `[_centralManager state]` returns `[CBManagerStateUnauthorized]` we could argue that the service is enabled but the application is not authorised. Unfortunately the [documentation](https://developer.apple.com/documentation/corebluetooth/cbmanagerstate?language=objc) is not really clear on this. 

Did you by change test this situation specifically?

**Review by mvanbeusekom — CHANGES_REQUESTED** [review]
The update to use a `completion` block looks really useful. The current implementation however does contain a bug (see comment).

Additionally I feel the `BluetoothPermissionStrategy.checkServiceStatus` might not cover all cases where the service is enabled.

File: permission_handler_apple/ios/Classes/strategies/BluetoothPermissionStrategy.m (hunk 1 above)
Comment (JeroenWeener) [reply permission_handler_apple/ios/Classes/strategies/BluetoothPermissionStrategy.m:43]: I also cannot find any documentation on `CBManagerStateUnauthorized`. Currently, I'm receiving `CBManagerStatePoweredOn` and `CBManagerStatePoweredOff`, when Bluetooth is enabled/disabled, respectively, regardless of Bluetooth permission.

We could opt for considering `CBManagerStateUnauthorized` as `ServiceStatusEnabled`. What do you think?

Comment (JeroenWeener) [reply permission_handler_apple/ios/Classes/strategies/BluetoothPermissionStrategy.m:43]: I have tested the behavior. It works as follows:

- If the application has Bluetooth permission, everything works. We receive either `CBManagerStatePoweredOn` or `CBManagerStatePoweredOff`, depending on the Bluetooth service status.
- If the application was explicitly not granted access to Bluetooth (permission denied), we receive `CBManagerStateUnauthorized`, regardless of the Bluetooth service status.
- If the application has not requested Bluetooth permission yet, and therefore the permission state is unkown, the application will ask for permission first, and then carry out the logic as described in the two points above.

It seems that the last point is only true since iOS 13. https://support.apple.com/en-us/HT210578: 'Starting in iOS 13, iPadOS 13, watchOS 6, and tvOS 13, an app must ask permission to use Bluetooth functions'.

So basically, we can only reliably determine the Bluetooth status if permission is granted. Additionally, even checking the Bluetooth service status will prompt the user for permission.

What do you think, @mvanbeusekom? Would this be okay for the feature, provided we document this behavior?

Comment (mvanbeusekom) [reply permission_handler_apple/ios/Classes/strategies/BluetoothPermissionStrategy.m:43]: Agreed, if we document this clearly this should be fine.

**Review by mvanbeusekom — APPROVED** [review]
LGTM
